core/controllers/admin.py

```diff
@@ -588,6 +588,45 @@ def _generate_dummy_explorations(
             raise Exception('Cannot generate dummy explorations in production.')
 
 
+class ExplorationsLatexSvgHandler(base.BaseHandler):
+    """Handler updating Explorations having math rich-text components with
+    math SVGs.
+
+    TODO(#10045): Remove this function once all the math-rich text components in
+    explorations have a valid math SVG stored in the datastore.
+    """
+
+    @acl_decorators.can_access_admin_page
+    def get(self):
+        latex_strings_to_exp_id_mapping = (
+            exp_services.get_batch_of_exps_for_latex_svg_generation())
+        self.render_json({
+            'latex_strings_to_exp_id_mapping': latex_strings_to_exp_id_mapping
+        })
+
+    @acl_decorators.can_access_admin_page
+    def post(self):
+        latex_to_svg_mappings = self.payload.get('latexMapping')
+        for exp_id, latex_to_svg_mapping_dict in latex_to_svg_mappings.items():
+            for latex_string in latex_to_svg_mapping_dict.keys():
+                svg_image = self.request.get(
+                    latex_to_svg_mappings[exp_id][latex_string]['latexId'])
+                if not svg_image:
+                    raise self.InvalidInputException(
+                        'SVG for LaTeX value %s in exploration %s is not '
+                        'supplied.' % (latex_string, exp_id))
+                latex_to_svg_mappings[exp_id][latex_string]['svg_file'] = (
+                    svg_image)
+
+        for exp_id in latex_to_svg_mappings.keys():
+            exp_services.update_exploration_with_math_svgs(
+                exp_id, latex_to_svg_mappings[exp_id])
+        self.render_json({
+            'number_of_explorations_updated': '%d' % (
```

Comment:
Will this message show up in the admin page? If so, that would be great.

Reply:
Yeah It will